### PR TITLE
bug(5015) pg doesn't support create table if not exists for 9.0 and below

### LINKED ISSUE #5015: Syntax error on sequelize.define()
I have a simple setup for Sequelize.
```javascript
var config = require('../config'),
    Sequelize = require('sequelize'),
    sequelize = new Sequelize(
        config.get('database'),
        config.get('username'),
        config.get('password'), {
            host: config.get('host'),
            port: config.get('dbport'),
            dialect: config.get('dialect'),
            pool: {
                max: 5,
                min: 0,
                idle: 10000
            }
        })
```
If i run an ordinary check `sequelize.authenticate()` it works fine and shows that there are no problems with connection. But if i try to define any model, `sequelize.sync();` throws an error: `Unhandled rejection SequelizeDatabaseError: syntax error at or near "NOT"`. It seems to me that there is a problem within sequelize or a driver. 
BTW I use PostgreSQL 9.0.13
Package dependencies are:
*  "pg": "^4.4.3",
* "pg-hstore": "^2.3.2",
* "sequelize": "^3.14.2",

### PR DESCRIPTION
**Pull request by sushantdhiman**
Fixes #5015

### CODE REVIEW

**Comment by sushantdhiman** [comment]
@janmeier *ping*

**Comment by sushantdhiman** [comment]
Hi @janmeier , Any improvements you want in this PR.

**Comment by mickhansen** [comment]
LGTM, maybe a unit test?

**Comment by sushantdhiman** [comment]
@mickhansen Added testes.

File: test/unit/sql/create-table.test.js [hunk 1]
```diff
@@ -27,6 +27,31 @@ describe(Support.getTestDialectTeaser('SQL'), function() {
         });
       });
     });
+    if(Support.getTestDialect() === 'postgres') {
```
Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: `current.dialect.name === 'postgres'` is the convention for unit tests (we're moving to that generally)

Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: please also add a space after `if`

File: test/unit/sql/create-table.test.js [hunk 2]
```diff
@@ -27,6 +27,31 @@ describe(Support.getTestDialectTeaser('SQL'), function() {
...
+        it('it will have IF NOT EXISTS for version 9.1 or above', function () {
+          modifiedSQL.sequelize.options.databaseVersion = '9.1.0';
+          var createTableQueryModified = sql.createTableQuery.bind(modifiedSQL);
+          expectsql(createTableQueryModified(FooUser.getTableName(), sql.attributesToSQL(FooUser.rawAttributes), { }), {
+            postgres: 'CREATE TABLE IF NOT EXISTS "foo"."users" ("id"   SERIAL , "mood" "foo"."enum_users_mood", PRIMARY KEY ("id"));'
+          });
+        });
+        it('it will have IF NOT EXISTS for default version', function () {
+          modifiedSQL.sequelize.options.databaseVersion = 0;
+          var createTableQueryModified = sql.createTableQuery.bind(modifiedSQL);
```
Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: Could this not be hoisted to modifiedSQL aswell?

File: test/unit/sql/create-table.test.js [hunk 3]
```diff
@@ -27,6 +27,31 @@ describe(Support.getTestDialectTeaser('SQL'), function() {
         });
       });
     });
+    if(Support.getTestDialect() === 'postgres') {
+      describe('offers Postgres support', function() {
```
Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: perhaps rename to `IF NOT EXISTS version check`

Comment (sushantdhiman) [inline test/unit/sql/create-table.test.js]: Should I rename to `Postgres IF NOT EXISTS version check`, as its a Postgres only check

Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: Nah, that is clear from the if around the test :)

Comment (mickhansen) [inline test/unit/sql/create-table.test.js]: Currently this only applies to postgres, but in the future it could apply to other dialects possibly.

**Comment by sushantdhiman** [comment]
This PR is squashed as well. Let me know if there are other improvements @mickhansen

**Comment by mickhansen** [comment]
Looks good, just waiting for CI

**Comment by mickhansen** [comment]
Thanks for your work @sushantdhiman

**Pull request merged**